synapse/handlers/cas_handler.py

```diff
@@ -240,20 +251,29 @@ async def _map_cas_user_to_matrix_user(
             user_agent: The user agent of the client making the request.
             ip_address: The IP address of the client making the request.
 
+        Raises:
+            MappingException: if there was an error while mapping some properties
+
         Returns:
              The user ID associated with this response.
         """
 
-        localpart = map_username_to_mxid_localpart(remote_user_id)
-        user_id = UserID(localpart, self._hostname).to_string()
-        registered_user_id = await self._auth_handler.check_user_exists(user_id)
+        async def cas_response_to_user_attributes(failures: int) -> UserAttributes:
+            """
+            Map from CAS attributes to user attributes.
+            """
+            localpart = map_username_to_mxid_localpart(remote_user_id)
 
-        # If the user does not exist, register it.
-        if not registered_user_id:
-            registered_user_id = await self._registration_handler.register_user(
-                localpart=localpart,
-                default_display_name=display_name,
-                user_agent_ips=[(user_agent, ip_address)],
-            )
+            # Append suffix integer if last call to this function failed to produce
+            # a usable mxid.
+            localpart += str(failures) if failures else ""
```

Comment:
CAS doesn't implement mapping providers, but it seemed the trivial thing to do here is just append the number as we do for the default OIDC/SAML mapping providers.